cjworkbench/i18n/views.py

```diff
@@ -23,4 +23,7 @@ def set_locale(request):
     locale = request.POST.get("new_locale")
     if is_supported(locale):
         request.session["locale_id"] = locale
+        if request.user.is_authenticated:
+            request.user.user_profile.locale_id = locale
+            request.user.user_profile.save()
```

Comment:
Never call `Model.save()` without arguments. It's race-prone.

Use `request.user.user_profile.save(update_fields=['locale_id'])`

(Django ORM slogan: "make easy tasks hard and hard tasks impossible.")